openedx_events/event_bus/__init__.py

```diff
@@ -116,11 +119,67 @@ def get_producer() -> EventBusProducer:
     If misconfigured, returns a fake implementation that can be called but does nothing.
     """
     return _try_load(
-        setting_name='EVENT_BUS_PRODUCER',
+        setting_name='EVENT_BUS_PRODUCER', args=(), kwargs={},
         expected_class=EventBusProducer, default=NoEventBusProducer(),
     )
 
 
+class EventBusConsumer(ABC):
+    """
+    Parent class for event bus consumer implementations.
+    """
+
+    @abstractmethod
+    def consume_indefinitely(self) -> None:
+        """
+        Consume events from a topic in an infinite loop.
+
+        Events will be converted into calls to Django signals.
+        """
+
+
+class NoEventBusConsumer(EventBusConsumer):
+    """
+    Stub implementation to "load" when no implementation is properly configured.
+    """
+
+    def consume_indefinitely(self) -> None:
+        """Do nothing."""
+
+
+# .. setting_name: EVENT_BUS_CONSUMER
+# .. setting_default: None
+# .. setting_description: String naming a callable (function or class) that can be called to create
+#   or retrieve an instance of EventBusConsumer when ``openedx_events.event_bus.make_single_consumer`` is
+#   called. The format of the string is a dotted path to an attribute in a module, e.g.
+#   ``some.module.path.EventBusImplementation``. See docstring of ``make_single_consumer`` for
+#   parameters. If setting is not supplied or the callable raises an exception or does not return
+#   an instance of EventBusConsumer, calls to the consumer will be ignored with a warning at startup.
+
+
```

Comment:
what are we documenting here?

Reply:
`EVENT_BUS_CONSUMER` at like 178 similar to `EVENT_BUS_PRODUCER` in line 105.

Reply:
`EVENT_BUS_PRODUCER` works the same way (accepting a dotted module string to a Callable) so I think we should stick to a parallel name.

Reply:
Updated!

Reply:
Btw, updating `EVENT_BUS_PRODUCER` will require changes in [edx-platform](https://github.com/search?q=repo%3Aopenedx%2Fedx-platform+EVENT_BUS_PRODUCER&type=code) as well.

Reply:
@robrap How strongly do you feel about a name including `CALLABLE`? I had been hoping to keep it short.

Reply:
We can leave as-is.

Reply:
Reverted.